test/get-group-tangle.test.js

```diff
@@ -0,0 +1,237 @@
+
+const GetGroupTangle = require('../lib/get-group-tangle')
+const Testbot = require('./helpers/testbot')
+const replicate = require('./helpers/replicate')
+
+test('get-group-tangle unit test', (t) => {
+  const name = `get-group-tangle-${Date.now()}`
+  const server = Testbot({ name })
+
+  server.tribes2.create(null, (err, group) => {
+    if (err) throw err
+
+    const getGroupTangle = GetGroupTangle(server)
+
+    getGroupTangle(group.id, (err, groupTangle) => {
+      if (err) throw err
+
+      const { root, previous } = groupTangle
+      const rootKey = group.root
+
+      pull(
+        server.db.query(where(author(server.id)), descending(), toPullStream()),
+        pull.map((m) => m.key),
+        pull.take(1),
+        pull.collect((err, keys) => {
+          if (err) throw err
+
+          t.deepEqual(
+            { root, previous },
+            { root: rootKey, previous: [keys[0]] },
+            'group add-member of admin should be the tip'
+          )
+
+          //  publishing to the group:
+          const content = {
+            type: 'memo',
+            root: rootKey,
+            message: 'unneccessary',
+            recps: [group.id],
+          }
+
+          server.tribes2.publish(content, (err, msg) => {
+            if (err) throw err
+
+            getGroupTangle(group.id, (err, { root, previous }) => {
+              if (err) throw err
+              t.deepEqual(
+                { root, previous },
+                { root: rootKey, previous: [msg.key] },
+                'adding message to root'
+              )
+
+              server.tribes2.publish(content, (err, msg) => {
+                if (err) throw err
+
+                getGroupTangle(group.id, (err, { root, previous }) => {
+                  if (err) throw err
+                  t.deepEqual(
+                    { root, previous },
+                    { root: rootKey, previous: [msg.key] },
+                    'adding message to tip'
+                  )
+                  server.close(true, t.end)
+                })
+              })
+            })
+          })
+        })
+      )
+    })
+  })
+})
+
+const n = 100
+test(`get-group-tangle-${n}-publishes`, (t) => {
+  const publishArray = new Array(n).fill().map((item, i) => i)
+  const server = Testbot()
+  let count = 0
+  server.tribes2.create(null, (err, data) => {
+    if (err) throw err
+
+    const groupId = data.id
+    pull(
+      pull.values(publishArray),
+      paraMap(
+        (value, cb) =>
+          server.tribes2.publish({ type: 'memo', value, recps: [groupId] }, cb),
+        4
+      ),
+      paraMap((msg, cb) => server.db.getMsg(msg.key, cb), 10),
+      pull.drain(
+        (m) => {
+          count += m.value.content.tangles.group.previous.length
+        },
+        (err) => {
+          t.error(err, 'no error')
+
+          // t.equal(count, n, 'We expect there to be no branches in our groupTangle')
```

Comment:
```suggestion
```

What was this assertion doing? (Is it from ssb-tribes1?)

Reply:
it was commented out already in tribes1 :shrug:

Reply:
Yeah, just delete it.